inc/vcf/compression.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2014-2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COMPRESSION_HPP
+#define COMPRESSION_HPP
+
+#include <iostream>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <boost/filesystem.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/iostreams/filter/bzip2.hpp>
+#include <boost/iostreams/filter/gzip.hpp>
+#include <boost/iostreams/filter/zlib.hpp>
+#include <boost/iostreams/filtering_stream.hpp>
+
+#include "util/string_utils.hpp"
+#include "vcf/report_writer.hpp"
+#include "vcf/string_constants.hpp"
+
+
+namespace ebi
+{
+  namespace vcf
+  {
+    /*
+     * method to get compression of a file.
+     * Params:
+     *  source - source_name
+     *  line - vector of first 5 characters or first line(if length of first line < 5) of file.
+     */
```

Comment:
Could the comments in this class be structured like Doxygen ones so they are appropriately generated if needed? See [this](https://github.com/EBIvariation/vcf-validator/blob/master/inc/vcf/file_structure.hpp#L260) for an example.